src/common/OutputDataSocket.cc

```diff
@@ -298,9 +298,7 @@ void OutputDataSocket::handle_connection(int fd)
 int OutputDataSocket::dump_data(int fd)
 {
   m_lock.Lock(); 
-  list<bufferlist> l;
-  l = data;
-  data.clear();
+  list<bufferlist> l = std::move(data);
```

Comment:
`data` is in a valid but unspecified state. `clear()` has no preconditions and it will return `data` to a known state.